### ISSUE TITLE
Linter for partially matched arguments

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2562]
In https://github.com/topepo/caret/pull/1361, I came across a lot of partially-matched argument calls, especially `seq(along=` and `seq(length=` (should be `seq(along.with=` and `seq(length.out=)`, respectively).

It would be good to have a linter for this. This will require having function definitions loaded, in which case we can use `match.call()` with `options(warnPartialMatchArgs=TRUE)` set to check for issues:

```r
c <- substitute(seq(along = x))
options(warnPartialMatchArgs = TRUE, warn=2L)
tryCatch(math.call(seq, c), error=\(c) "lint")
#  [1] "lint"
```

The key is resolving `seq` generically for any call name in the linted code.

**Comment by MichaelChirico** [comment]
Some exploratory code for potential implementation:

```r
xml = source_expression$full_xml_parsed_content
all_call_expr = xml_find_all(xml, "//SYMBOL_FUNCTION_CALL/parent::expr/parent::expr")
# xml2lang() still not robust enough to generically convert <expr>...</expr> --> lang
all_call_lang = Filter(length, lapply(all_call_expr, \(x) tryCatch(xml2lang(x), error=\(c) NULL))
# drop primitives, e.g. match.call(c, quote(c(1))) fails; also no hope to resolve all calls generally,
#   e.g. nested / anonymous functions, so just skip those
all_calls=Filter(\(e) tryCatch(!is.primitive(eval(e[[1]])), error=\(c) FALSE), all_calls)
lapply(all_calls, \(e) match.call(eval(e[[1]]), e))
```

**Comment by ben-schwen** [comment]
As we noticed in https://github.com/Rdatatable/data.table/issues/6855 partial arguments can still be in CRAN packages in non executable code such as vignettes or other documentation files, so still might be worth to find them on CRAN packages
